federation/apis/federation/types.go

```diff
@@ -108,3 +108,27 @@ type ClusterList struct {
 	// List of Cluster objects.
 	Items []Cluster `json:"items"`
 }
+
+// Temporary/alpha stuctures to support custom replica assignments within FederatedReplicaSet.
+
+// A set of preferences that can be added to federated version of ReplicaSet as a json-serialized annotation.
+// The preferences allow the user to express in which culsters he wants to put his replicas within the
+// mentiond FederatedReplicaSet.
+type FederatedReplicaSetPreferences struct {
+	// A maping between cluser names and preferences regarding local replicasets in these clusters.
+	// "*" applies to all clusters if explicint mapping is not provided. It is assumed that cluster
+	// should not have any replicas scheduled.
+	Clusters map[string]LocalReplicaSetPreferences `json:"clusters,omitempty"`
+}
+
+// Preferences regaring number of replicas assigned to a local replicaset within federated replica set.
+type LocalReplicaSetPreferences struct {
+	// Minimum number of replicas that should be assigned to this Local ReplicaSet. 0 by default.
+	MinReplicas int64 `json:"maxReplicas,omitempty"`
+
+	// Maximum number of replicas that should be assigned to this Local ReplicaSet. Unbounded if no value provided (default).
+	MaxReplicas *int64 `json:"maxReplicas,omitempty"`
+
+	// A number expressing the preference to put an additional replica to this LocalReplicaSet. 0 by default.
+	Weight int64
```

Comment:
AFAIR Brian is against any floats in the api and we don't have it anywhere. The problem with floats is that two similarly looking/computed floats may in fact be different causing various issues in comparisons.